poutyne/framework/callbacks/wandb_logger.py

```diff
@@ -0,0 +1,256 @@
+
+            if wandb.run is None:
+                self.run = wandb.init(**self._wandb_init)
+            else:
+                warnings.warn(
+                    "There is already a wandb run experience running. This callback will reuse this run. If you want "
+                    "to start a new one stop this process and call `wandb.finish()` before starting again."
+                )
+                self.run = wandb.run
+        else:
+            self.run = experiment
+
+        self.run.config.update({"run_id": self.run.id})
+        self.log_gradient_frequency = log_gradient_frequency
+        self.training_batch_shape = training_batch_shape
+        self.checkpoints_path = checkpoints_path
+        if initial_artifacts_paths is not None:
+            self._log_artifacts(initial_artifacts_paths, name="Initial-artifacts", artifact_type="Miscellaneous")
+
+    def _watch_gradient(self) -> None:
+        """
+        Activate wandb gradient watching.
+        """
+        self.run.watch(self.model.network, log="all", log_freq=self.log_gradient_frequency)
+
+    def _save_architecture(self) -> None:
+        """
+        Save architecture.
+        """
+        dummies_batch = torch.randn(self.training_batch_shape)
+        save_path = self.run.dir + "/" + self.run.name + "_model.onnx"
+        torch.onnx.export(self.model.network, dummies_batch, save_path)
+        self.run.save(save_path)
+
+    def on_train_begin(self, logs: Dict):
+        super().on_train_begin(logs)
+        if self.log_gradient_frequency is not None:
+            self._watch_gradient()
+
+        if self.training_batch_shape is not None:
+            self._save_architecture()
+
+    def log_config_params(self, config_params: Dict) -> None:
+        """
+        Args:
+            config_params Dict:
+                Dictionary of config parameters of the training to log, such as number of epoch, loss function,
+                optimizer etc.
+        """
+        self.run.config.update(config_params)
+
+    def _on_train_batch_end_write(self, batch_number: int, logs: Dict) -> None:
+        """
+        Log the batch metric.
+        """
+        if self.batch_granularity:
+            train_metrics = {key: value for (key, value) in logs.items() if "val_" not in key}
+            train_metrics = {"training": {"batch": train_metrics}}
+            self._log_metrics(train_metrics)
+
+    def _on_epoch_end_write(self, epoch_number: int, logs: Dict) -> None:
+        """
+        Log the epoch metric.
+        """
+        train_metrics = {key: value for (key, value) in logs.items() if "val_" not in key}
+        val_metrics = {key.replace("val_", ""): value for (key, value) in logs.items() if "val_" in key}
+        learning_rate = self._get_current_learning_rates()
+
+        if self.batch_granularity:
+            train_metrics = {"training": {"epoch": train_metrics}}
+        else:
+            train_metrics = {"training": train_metrics}
```

Comment:
Shouldn't you do the same thing for validation below?

Reply:
Yes I changed it.